paddle/math/Matrix.h

```diff
@@ -997,8 +997,8 @@ class Matrix : public BaseMatrix {
     LOG(FATAL) << "Not implemeted";
   }
 
-  virtual void contextProjectionForward(MatrixPtr input,
-                                        MatrixPtr weight,
+  virtual void contextProjectionForward(Matrix& input,
+                                        Matrix* weight,
```

Comment:
The main motivation to use Matrix * here is that we need to compare "if (weight)" in Matrix.cpp. It might be convenient to use raw pointer here instead of reference. What do you suggest?